server/base/src/main/java/org/apache/accumulo/server/zookeeper/TransactionWatcher.java

```diff
@@ -25,7 +25,9 @@
 import java.util.Set;
 import java.util.concurrent.Callable;
 import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.Lock;
 
+import com.google.common.util.concurrent.Striped;
```

Comment:
It looks like this is `@Beta`.

Reply:
Just as a note, it appears it has been in beta since 2012, v 13.0 (now v 30.0 is released). I don't expect it to leave beta anytime soon.

Reply:
It doesn't seem that long ago since we were using version 11 of Guava, because that's what Hadoop was using. Guava moves fast. The beta methods can change quickly and can quickly become incompatible with other libraries that require specific versions of Guava. I still think this is unsafe.

More importantly, if there's no observable performance benefit, it's not worth the risk.

Reply:
I would avoid Guava striped locks because its beta.  I have used this pattern w/o Guava using an array of locks like : 

https://github.com/apache/accumulo/blob/46ee51d0c6afe9910873f55653017b91c45a345a/core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/lru/SynchronousLoadingBlockCache.java#L38

Reply:
Use of Beta can be dangerous as we have been burned previously. I did some work to remove uses of Beta code in Accumulo. See https://issues.apache.org/jira/browse/ACCUMULO-4702
I would listen to this important declaration:

> If your code is a library itself (i.e. it is used on the CLASSPATH of users outside your own control), you should not use beta API